2011-03-01  Aurelien Jarno  <aurelien@aurel32.net>

	* sysdeps/s390/s390-64/Makefile(utf8-utf32-z9.c): pass 
	-march=z9-109 to the assembler.
	(utf16-utf32-z9.c): Likewise.
	(utf8-utf16-z9.c): Likewise.

diff --git a/sysdeps/s390/s390-64/Makefile b/sysdeps/s390/s390-64/Makefile
index 1814f37..bb0b62c 100644
--- a/sysdeps/s390/s390-64/Makefile
+++ b/sysdeps/s390/s390-64/Makefile
@@ -16,12 +16,15 @@ ISO-8859-1_CP037_Z900-map := gconv.map
 
 UTF8_UTF32_Z9-routines := utf8-utf32-z9
 UTF8_UTF32_Z9-map := gconv.map
+CFLAGS-utf8-utf32-z9.c += -Wa,-march=z9-109
 
 UTF16_UTF32_Z9-routines := utf16-utf32-z9
 UTF16_UTF32_Z9-map := gconv.map
+CFLAGS-utf16-utf32-z9.c += -Wa,-march=z9-109
 
 UTF8_UTF16_Z9-routines := utf8-utf16-z9
 UTF8_UTF16_Z9-map := gconv.map
+CFLAGS-utf8-utf16-z9.c += -Wa,-march=z9-109
 
 s390x-iconv-modules = ISO-8859-1_CP037_Z900 UTF8_UTF16_Z9 UTF16_UTF32_Z9 UTF8_UTF32_Z9
